enumerable_methods.rb

```diff
@@ -0,0 +1,159 @@
+# frozen_string_literal: true
+
+module Enumerable # rubocop:disable Metrics/ModuleLength
+  # my_each
+  def my_each
+    return to_enum unless block_given?
+
+    i = 0
+    while i < size
+      yield(self[i])
+      i += 1
+    end
+  end
+
+  # my_each_with_index
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    i = 0
+    while i < size
+      yield(self[i], i)
+      i += 1
+    end
+  end
+
+  # my_select
+  def my_select
+    return to_enum unless block_given?
+
+    arr = []
+    my_each do |a|
+      arr.push(a) if yield(a)
+    end
+    arr
+  end
+
+  # my_all
+  def my_all?(arg = nil) # rubocop:disable Layout/TrailingWhitespace
+    if block_given?
+      my_each { |a| return false unless yield(a) }
+      return true
+    end
+    if arg.class == Regexp
+      my_each { |a| return false unless a.match(arg) }
+      return true
+    elsif arg.class == Class
+      my_each { |a| return false unless a.class == arg || a.class.superclass == arg }
+      return true
+    elsif arg.nil? && !block_given? && !empty?
+      return false
+    end
+    true
+  end
```

Comment:
**[REQUIRED]**
* When no block or argument is given returns true when none of the collection members are false or nil.
![all](https://user-images.githubusercontent.com/44072711/68567772-dd39a680-0427-11ea-9769-90ac77fe2cc4.png).
* when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern.
![ptt](https://user-images.githubusercontent.com/44072711/68568183-ed05ba80-0428-11ea-8a30-0a511b330734.png)